apps/api/src/music/music.controller.ts

```diff
@@ -32,44 +34,109 @@ export class MusicController {
 
   @Post()
   @ApiOperation({ description: 'Create music' })
+  async create(@Body() data: CreateMusicDto) {
+    const result = await this.musicService.create({
+      ...data,
+    });
+
+    this.logger.debug(`MUSIC CREATE | Recieved data client:\n${result}`);
+    return {
+      message: 'Music was successfully created',
+      success: true,
+      data: {
+        date: result.createdAt,
+        id: result.id,
+      },
+    };
+  }
+
+  @Post(':id/upload')
+  @ApiOperation({ description: 'Upload audio file for a specific music' })
   @UseInterceptors(
-    FileInterceptor('file', {
+    FileInterceptor('musicAudioFileName', {
       storage: diskStorage({
-        destination: multerDiskStorageDestination,
+        destination: multerDiskStorageMusicAudioFileDestination,
         filename: multerDiskStorageFilename,
       }),
       fileFilter: audioFileFilter,
     })
   )
-  async create(
-    @Body() data: CreateMusicDto,
+  async uploadMusic(
+    @Param('id') id: string,
     @UploadedFile()
-    file: Express.Multer.File
+    musicAudioFileName: Express.Multer.File
   ) {
-    if (!file) {
+    if (!musicAudioFileName) {
+      this.logger.log(musicAudioFileName);
       throw new BadRequestException('No file uploaded');
     }
 
-    const result = await this.musicService.create({
-      ...data,
-      fileName: file.filename,
+    const result = await this.musicService.update(id, {
+      musicAudioFileName: musicAudioFileName.filename,
     });
+
     this.logger.debug(`MUSIC CREATE | Recieved data client:\n${result}`);
-    this.logger.verbose(`MUSIC CREATE | Music file uploaded: ${file.filename}`);
+    this.logger.verbose(
+      `MUSIC CREATE | Music file uploaded: ${musicAudioFileName.filename}`
+    );
     return {
       message: 'Music was successfully created',
       success: true,
       data: {
-        date: result.createdAt,
-        fileName: file.filename,
+        musicAudioFileName: musicAudioFileName.filename,
         id: result.id,
       },
     };
   }
 
-  @Get('file/:id')
-  public async getFile(@Param('id') id: string) {
-    return this.musicService.getMusicFile(id);
+  @Post(':id/cover/upload')
```

Comment:
```suggestion
  @Patch(':id/cover')
```

Reply:
Fair enough, that's good standard to adapt.

Reply:
This idea can be followed from #19